internal/testrunner/runners/system/runner.go

```diff
@@ -650,6 +650,14 @@ func writeSampleEvent(path string, doc common.MapStr) error {
 
 func validateFields(docs []common.MapStr, fieldsValidator *fields.Validator, dataStream string) error {
 	var multiErr multierror.Error
+
+	// prevents panic and performs cleanup/teardown for cases where
+	// no hits were found for a data stream
+	if len(docs) == 0 {
+		multiErr = append(multiErr, fmt.Errorf("no documents found in %s data stream", dataStream))
+		return multiErr
```

Comment:
I think that to make it correct, the code execution must reach line 676 (`return testrunner.ErrTestCaseFailed{`) to convert the multiErr.

I would leave the `multiErr = append(...)` in line 657, but wrap for-loop with if-clause.

Reply:
Thanks for pointing that out @mtojek. Sorry missed that. I've made the change.